plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/ProgramConverter.kt

```diff
@@ -66,30 +68,35 @@ class ProgramConverter(val session: FirSession) : ProgramConversionContext {
 
     private fun processFunction(symbol: FirNamedFunctionSymbol, body: FirBlock?): MethodSignatureEmbedding {
         val signature = embedSignature(symbol)
+
+        val methodCtx = object : MethodConversionContext, ProgramConversionContext by this {
+            override val signature: MethodSignatureEmbedding = signature
+
+            private var nextAnonVarNumber = 0
+            override fun newAnonVar(type: TypeEmbedding): VariableEmbedding =
+                VariableEmbedding(AnonymousName(++nextAnonVarNumber), type)
+
+            val contractPostconditions = symbol.resolvedContractDescription?.effects?.map {
+                it.effect.accept(ContractDescriptionConversionVisitor(), this)
+            } ?: emptyList()
+
+            override val preconditions: List<Exp> = signature.params.flatMap { it.invariants() }
+            override val postconditions: List<Exp> =
+                signature.params.flatMap { it.invariants() } + signature.params.flatMap { it.dynamicInvariants() } + signature.returnVar.invariants() + contractPostconditions
```

Comment:
Can this be done by splitting it out in a different interface and then using `by` to implement it?